### PR TITLE
(PC-21770) fix(ci): slack should notify if a error arrived

### PR DESCRIPTION
**Pull request by mlachevre-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-22460

## Purpose

Slack should notify if the previous job failed currently it's not the case because slack is one of the step of the job failing.

### CODE REVIEW

File: .github/workflows/dev_on_workflow_environment_deploy.yml
Comment (jroussel-pass) [inline .github/workflows/dev_on_workflow_environment_deploy.yml:1]: Je pense que 
```
SLACK_BOT_TOKEN:passculture-metier-ehp/passculture-app-native-slack-token
```
à la ligne 52 n'est plus utile

File: .github/workflows/dev_on_workflow_environment_ios_deploy.yml
Comment (jroussel-pass) [inline .github/workflows/dev_on_workflow_environment_ios_deploy.yml:1]: Pareil mais ligne 57

File: .github/workflows/dev_on_workflow_web_deploy.yml
Comment (jroussel-pass) [inline .github/workflows/dev_on_workflow_web_deploy.yml:1]: Pareil mais ligne 49

**Review by jroussel-pass — COMMENTED** [review]
Sinon le reste me parait ok.

File: .github/workflows/dev_on_workflow_environment_deploy.yml
Comment (mlachevre-pass) [reply .github/workflows/dev_on_workflow_environment_deploy.yml:1]: bien vu

File: .github/workflows/dev_on_workflow_environment_ios_deploy.yml
Comment (mlachevre-pass) [reply .github/workflows/dev_on_workflow_environment_ios_deploy.yml:1]: bien vu

**Review by jroussel-pass — APPROVED** [review]

**Pull request merged**